lib/chef/http/authenticator.rb

```diff
@@ -104,6 +152,70 @@ def load_signing_key(key_file, raw_key = nil)
         raise Chef::Exceptions::InvalidPrivateKey, msg
       end
 
+      # takes no parameters. Checks for the password in the registry and returns it if there, otherwise returns false
+      def self.get_cert_password
+        @win32registry = Chef::Win32::Registry.new
+        path = "HKEY_LOCAL_MACHINE\\Software\\Progress\\Authentication"
+        # does the registry key even exist?
+        present = @win32registry.get_values(path)
+        if present.nil? || present.empty?
+          raise Chef::Exceptions::Win32RegKeyMissing
+        end
+
+        if present.map { |h| h[:name] }[0] == "PfxPass"
+          present.map { |h| h[:data] }[0].to_s
+        end
```

Comment:
```suggestion
        present_info = present.map { |h| h[:name] }
        if present_info[0] == "PfxPass"
          present[0]
        end
```

Reply:
That statement has a bug in it - present_info[0] on line 167 only returns the name "PfxPass". However, I took your advice to heart and wrote that out more clearly. I did this:

```Ruby
        present.each do |secret|
          if secret[:name] == "PfxPass"
            secret[:data]
          else
            raise Chef::Exceptions::Win32RegKeyMissing
          end
        end
```

Reply:
I added the exception in case there was a block of crap at that hive level that did not include the PfxPass object